src/main/java/tutorspet/logic/util/AttendanceRecordListUtil.java

```diff
@@ -0,0 +1,155 @@
+     * {@code Attendance} for the {@code targetStudent} in the {@code targetWeek} in the
+     * {@code targetAttendanceRecordList}.
+     *
+     * @throws CommandException if the {@code targetWeek} does not exist in the {@code targetAttendanceRecordList} or
+     * if there does not exist an existing {@code Attendance} for the {@code targetStudent}.
+     */
+    public static AttendanceRecordList editAttendanceInAttendanceRecordList(
+            AttendanceRecordList targetAttendanceRecordList,
+            Student targetStudent, Week targetWeek,
+            Attendance attendanceToSet) throws CommandException {
+        requireAllNonNull(targetAttendanceRecordList, targetStudent, targetWeek, attendanceToSet);
+
+        if (!targetAttendanceRecordList.isWeekContained(targetWeek)) {
+            throw new CommandException(MESSAGE_INVALID_WEEK);
+        }
+
+        AttendanceRecord targetAttendanceRecord = targetAttendanceRecordList.getAttendanceRecord(targetWeek);
+
+        AttendanceRecord editedWeekAttendanceRecord =
+                setAttendance(targetAttendanceRecord, targetStudent, attendanceToSet);
+
+        List<AttendanceRecord> modifiedAttendanceRecordList =
+                new ArrayList<>(targetAttendanceRecordList.getAttendanceRecordList());
+        modifiedAttendanceRecordList.set(targetWeek.getZeroBasedWeekIndex(), editedWeekAttendanceRecord);
+        return new AttendanceRecordList(modifiedAttendanceRecordList);
+    }
+
+    /**
+     * Returns an {@code AttendanceRecordList} where the {@code Attendance} for the {@code targetStudent} in the
+     * {@code targetWeek} in the {@code targetAttendanceRecordList} has been removed.
+     *
+     * @throws CommandException if the {@code targetWeek} does not exist in the {@code targetAttendanceRecordList} or
+     * if there does not exist an existing {@code Attendance} for the {@code targetStudent}.
+     */
+    public static AttendanceRecordList removeAttendanceFromAttendanceRecordList(
+            AttendanceRecordList targetAttendanceRecordList,
+            Student targetStudent,
+            Week targetWeek) throws CommandException {
+        requireAllNonNull(targetAttendanceRecordList, targetStudent, targetWeek);
+
+        if (!targetAttendanceRecordList.isWeekContained(targetWeek)) {
+            throw new CommandException(MESSAGE_INVALID_WEEK);
+        }
+
+        AttendanceRecord targetAttendanceRecord = targetAttendanceRecordList.getAttendanceRecord(targetWeek);
+
+        AttendanceRecord editedWeekAttendanceRecord =
+                AttendanceRecordUtil.removeAttendance(targetAttendanceRecord, targetStudent);
+
+        List<AttendanceRecord> modifiedAttendanceRecordList =
+                new ArrayList<>(targetAttendanceRecordList.getAttendanceRecordList());
+        modifiedAttendanceRecordList.set(targetWeek.getZeroBasedWeekIndex(), editedWeekAttendanceRecord);
```

Comment:
Should you abstract these 2 lines out into a private method? These 2 lines are repeated in the above methods.